crates/blockchain-tree/src/block_buffer.rs

```diff
@@ -76,11 +73,10 @@ impl BlockBuffer {
         self.earliest_blocks.entry(block.number).or_default().insert(hash);
         self.blocks.insert(hash, block);
 
-        if let Some((evicted_hash, _)) = self.lru.push(hash, ()).filter(|(b, _)| *b != hash) {
-            // evict the block if limit is hit
-            if let Some(evicted_block) = self.remove_block(&evicted_hash) {
-                // evict the block if limit is hit
-                self.remove_from_parent(evicted_block.parent_hash, &evicted_hash);
+        if self.lru.insert(hash, ()) {
+            // Evict the block if limit is hit
+            if let Some(evicted_block) = self.remove_block(&hash) {
+                self.remove_from_parent(evicted_block.parent_hash, &hash);
```

Comment:
Curious if you think these changes are okay. I found myself wanting the `insert_and_get_evicted` method that is on the `LruCache`, but it wasn't implemented for the `LruMap`. Given my understanding this should achieve the same thing.

Reply:
Looks like this isn't quite right as it's causing some tests to fail.